### DISCUSSION TITLE
Clock fractional offsets in GOCE GPS data

### DISCUSSION THREAD

**Discussion by ParisaShafiei** [groops-devs/groops#231] [Q&A]
Dear developers, 

I have a question for you about precise orbit determination of the GOCE satellite. 

Given the work of [Bock et al., 2011](https://link.springer.com/article/10.1007/s00190-011-0484-9), it is mentioned that **the time tagging of the GOCE GPS data is special and as the internal clock is not steered to integer seconds, the observation epochs have fractional offsets.** 

I am wondering how this issue can be addressed in processing the kinematic orbits of this satellite. I think these fractional offsets can be problematic in comparing the kinematic orbit with the reduced dynamic as they are given in integer seconds and also later on it should be addressed in gravity field recovery based on hl-SST. 

While Bock et al., 2011 mentioned that they applied a simple spline interpolation to interpolate the GPS phase observations to integer seconds, I am wondering 1) how I can apply this in GROOPS and 2) if any similar or different corrections are applied in [Zehentner, N., Mayer-Gürr, T. Precise orbit determination based on raw GPS ](https://link.springer.com/article/10.1007/s00190-015-0872-7)?

Can you please give me an idea over it?

Like always, thank you for your time and support. 

Looking forward to hearing from you.

With kind regards, 
Parisa

**Comment by tmayerguerr** [comment]
Hi Parisa,

Sorry for my late answer.  I would do it the other way around:

- You have to interpolate the orbit and the starCamera data to the gnssReceiver data with [InstrumentResample](https://groops-devs.github.io/groops/html/InstrumentResample.html). The timeSeries should be set to the epochs of the gnssReceiver ([timeSeries->instrument](https://groops-devs.github.io/groops/html/timeSeriesType.html#instrument)).
- In [GnssProcessing](https://groops-devs.github.io/groops/html/GnssProcessing.html), the [timeSeries](https://groops-devs.github.io/groops/html/timeSeriesType.html) should be set also the same epochs (timeSeries->instrument)

Best regards
Torsten
